packages/core/src/js/integrations/devicecontext.ts

```diff
@@ -83,7 +84,8 @@ async function processEvent(event: Event): Promise<Event> {
     ? native['breadcrumbs'].map(breadcrumbFromObject)
     : undefined;
   if (nativeBreadcrumbs) {
-    event.breadcrumbs = nativeBreadcrumbs;
+    const maxBreadcrumbs = getClient()?.getOptions().maxBreadcrumbs ?? 100; // Default is 100.
+    event.breadcrumbs = nativeBreadcrumbs.concat(event.breadcrumbs || []).slice(0, maxBreadcrumbs);
```

Comment:
h: We have to ensure we keep the last `maxBreadcrumbs` (the last 100). Please also add a test that capture this edge case.

Reply:
Good point @krystofwoldrich 👍 
I've updated the logic and tests with c4246d789b690da34fb47056f38c63e25064a9ee